cpp/src/copying/copy.cu

```diff
@@ -188,16 +187,20 @@ std::unique_ptr<column> scatter_gather_based_if_else(cudf::column_view const& lh
                                                scatter_map.begin(),
                                                is_left);
 
+  auto gather_map =
+    column_view(data_type{type_to_id<size_type>()},
+                static_cast<size_type>(thrust::distance(scatter_map.begin(), scatter_map_end)),
+                scatter_map.data());
```

Comment:
This shouldn't be needed anymore?

Reply:
The size parameter value may not match the span's size.

Reply:
Oh, I would just resize the `device_uvector` in that case (resizing smaller will not incur any copies/allocations). 

```suggestion
   scatter_map.resize(std::distance(scatter_map.begin(), scatter_map_end));
```

Reply:
`scatter_map` should probably be renamed `gather_map` too.